Move Path::Pattern factories into test helper

composerinteralia opened the pull request:
`Pattern#from_string` was introduced in bb207ea7 to support creating
patterns from strings in tests (removing a conditional in the initialize
method that had been there only for the sake of those tests).

`Pattern#build` was introduced in 947ebe9a, and was similarly used only
in tests.

My understanding is that [`Mapping#path`][path] is the only place where
we initialize a `Path::Pattern` in library code.

Since these two methods appear to be used only in tests, this
commit moves them out of the class and into a test helper.

My reasoning for doing this is that I am doing some performance work
that may involve a modification to how `Path::Pattern`s get initialized,
and I will have more confidence in that work if I know for sure that
these methods are test helpers that can be modified freely.

[path]: https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/routing/mapper.rb#L192-L194

cc @eugeneius (I am tagging you because you seem interested in these Journey PRs, but can stop doing that if you are getting too many notifications)

eugeneius commented:
I am indeed interested in these PRs, feel free to 🔔 me on them!

This change looks fine to me, but as there isn't an immediate benefit and it's kind of an architectural decision, I'd like to see if @tenderlove has an opinion first.

composerinteralia commented:
Yeah, that makes sense. For full context, https://github.com/rails/rails/compare/master...composerinteralia:journey-performance-improvements is what I am working on. My main reason for opening this PR is that I was uncomfortable with defaulting to `true` on https://github.com/rails/rails/compare/master...composerinteralia:journey-performance-improvements#diff-67940fe9af75d2f6ec9596b2d13998c7R16 if this code ever gets used anywhere except in tests.

tenderlove commented on actionpack/test/support/path_helper.rb:
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module PathHelper
```
Can you put this module in a namespace?

composerinteralia replied:
Done!

tenderlove reviewed (approved):
LGTM. I left one comment, but it's not a blocker to merge this PR.  Thanks for working on this!

composerinteralia committed:
Move Path::Pattern factories into test helper

`Pattern#from_sting` was introduced in bb207ea7 to support creating
patterns from strings in tests (removing a conditional in the initialize
method that had been there only for the sake of those tests).

`Pattern#build` was introduced in 947ebe9a, and was similarly used only
in tests.

My understanding is that [`Mapping#path`][path] is the only place where
we initialize a `Path::Pattern` in library code.

Since these two methods appear to be used only in tests, this
commit moves them out of the class and into a test helper.

My reasoning for doing this is that I am doing some performance work
that may involve a modification to how `Path::Pattern`s get initialized,
and I will have more confidence in that work if I know for sure that
these methods are test helpers that can be modified freely.

[path]: https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/routing/mapper.rb#L192-L194

eugeneius commented:
Thank you both!
